### DISCUSSION TITLE
Any way to dump out information about what your script knows?

### DISCUSSION THREAD

**Discussion by ckipp01** [VirtusLab/scala-cli#1005] [Ideas]
## The problem
Is there a way for scala-cli to "dump" out information that it knows about it's "build definition" or things like what deps are being used? The background is that at work we've slowly been playing around with replacing some of our minimal sbt projects with scala-cli projects to avoid a bunch of boilerplate, but one of the requirements we have at work (and I imagine others in the industry might have) is to be able to do things like license checks on projects. Other "build" tools provide plugins to do stuff like this, but I imagine we don't want to get into that arena with scala-cli, however it'd be amazing if there was a way to "dump out" what scala-cli knows about the build. It knows it all, but doesn't expose any of it meaning that in situations like ours at work, we can't use it, because we can't dump out the deps that are being used to check the licenses.

## Potential solution
Since I'm assuming there is no desire to create a plugin infrastructure, it'd be amazing to provide a like `scala-cli dump` command or something that would dump out what it knows in a json file or something so that other tools could pick that up and do stuff with it. For our usecase I can imagine being able to run this command and then have a script do whatever checks we need on that information. Would something like this be doable if there isn't already a way to expose this info?

An adding benefit of doing this is that it also unlock the question I had in https://github.com/VirtusLab/scala-cli/discussions/549.

**Comment by alexarchambault** [comment]
Scala CLI uses internally a few case classes, such as [`BuildOptions`](https://github.com/VirtusLab/scala-cli/blob/310e9edbeebe7de5c2ff142bf02dd888f4dcd2a2/modules/options/src/main/scala/scala/build/options/BuildOptions.scala), [`Inputs`](https://github.com/VirtusLab/scala-cli/blob/310e9edbeebe7de5c2ff142bf02dd888f4dcd2a2/modules/build/src/main/scala/scala/build/Inputs.scala), [`Artifacts`](https://github.com/VirtusLab/scala-cli/blob/310e9edbeebe7de5c2ff142bf02dd888f4dcd2a2/modules/options/src/main/scala/scala/build/Artifacts.scala), or [`Sources`](https://github.com/VirtusLab/scala-cli/blob/310e9edbeebe7de5c2ff142bf02dd888f4dcd2a2/modules/build/src/main/scala/scala/build/Sources.scala) that we could definitely dump in JSON files on demand. These hold all that Scala CLI knows about the project. Support for dumping that could be added IMO.
